library/src/main/java/com/crossbowffs/remotepreferences/RemotePreferenceProvider.java

```diff
@@ -143,13 +143,21 @@ public boolean onCreate() {
             if (file.isDeviceProtected() && Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
                 context = context.createDeviceProtectedStorageContext();
             }
-            SharedPreferences prefs = context.getSharedPreferences(file.getFileName(), Context.MODE_PRIVATE);
+            SharedPreferences prefs = getSharedPreferences(context, file.getFileName());
             prefs.registerOnSharedPreferenceChangeListener(this);
             mPreferences.put(file.getFileName(), prefs);
         }
         return true;
     }
 
+    /**
+     * Generate {@link SharedPreferences} to store the key-value data.
+     * Override this method to provide a custom implementation of {@link SharedPreferences}.
```

Comment:
```suggestion
     * Generate {@link SharedPreferences} to store the key-value data.
     * Override this method to provide a custom implementation of {@link SharedPreferences}.
     *
     * @param context The context that should be used to get the preferences object.
     * @param prefFileName The name of the preference file.
     * @return An object implementing the {@link SharedPreferences} interface.
```

Mind adding javadoc for the parameters? Normally I hate writing obvious docs but the rest of the file does it in this style :p

Reply:
As you wish.